### DISCUSSION TITLE
blog: new post notification?

### DISCUSSION THREAD

**Discussion by kcgthb** [squidfunk/mkdocs-material#6083] [Ideas]
Hi,

I was wondering if the idea of having a way to notify readers about new blog entries had already been explored?

Something like a visual clue/badge/icon that could be displayed somewhere to let users know that a new blog entry had been posted. Maybe even track read/unread status via cookies?

**Comment by squidfunk** [comment]
Thanks for suggesting! Once https://github.com/squidfunk/mkdocs-material/issues/5074 is implemented, you could just embed your latest blog posts on your homepage? I'm not sure I'd want a notification/popup that the user has to click away. We try to minimize those, as this is rather a dark pattern, shoving the user things in the face, IMHO. Other than that, reading state is something that could be implemented, but it's very fragile, as we'd use local storage which cannot be considered persistent, and it will not sync across devices. This means that the user would see content as unread again when changing the device.

**Reply by squidfunk** [reply]
For new blog post notification there's [RSS](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#rss), and I think there's also a [mkdocs-newsletter](https://lyz-code.github.io/mkdocs-newsletter/) plugin, but I haven't tried it yet.

**Comment by kcgthb** [comment]
Thanks @squidfunk !

I like RSS, and embedding blog posts in the homepage would definitely be a possibility. But for new posts notifications, I was thinking more about a little visual clue, something subtle, like those unread badges that are pretty popular in Changelog-as-a-Service tools (see  example below):
![image](https://github.com/squidfunk/mkdocs-material/assets/186807/c897cc8a-7146-4299-9c68-cb959ead8183)

**Reply by squidfunk** [reply]
How do those services persist read state?

**Reply by kcgthb** [reply]
I assume via local storage / cookies, but it's likely not cross-devices either.

**Reply by squidfunk** [reply]
Yeah, so honestly, I'm not sure how this should work without being too annoying. Local storage must be considered volatile and I think it is okay to use for something like an announcement bar, but new post notifications are not something we're considering right now. MkDocs builds a static site that respects privacy, and such a notification would need some storage other than local storage for tracking read state reliably. However, this would require opt-in from the user and/or track the user without his consent.
